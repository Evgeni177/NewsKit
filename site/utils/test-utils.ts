import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
