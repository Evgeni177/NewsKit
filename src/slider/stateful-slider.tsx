import React, {useState} from 'react';
import {SliderProps, StatefulSliderProps} from './types';
import {Slider} from './slider';

export const StatefulSlider: React.FC<StatefulSliderProps> = ({
  values,
  ...props
}) => {
  const [stateValues, setValues] = useState(values);
  const onChange: SliderProps['onChange'] = newValues => {
    setValues(newValues);
    if (props.onChange) {
      props.onChange(newValues);
    }
  };
  return <Slider {...props} values={stateValues} onChange={onChange} />;
};
