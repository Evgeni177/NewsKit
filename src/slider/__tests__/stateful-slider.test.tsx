import {fireEvent} from '@testing-library/react';
import {renderWithTheme} from '../../test/test-utils';
import {StatefulSlider} from '../stateful-slider';

describe('StatefulSlider', () => {
  test('holds the slider values in state', () => {
    const slider = renderWithTheme(StatefulSlider, {
      values: [50],
      min: 25,
      max: 75,
      step: 1,
    });

    // Simulate pressing page down to lower value by 10
    const thumb = slider.getByRole('slider');
    fireEvent.keyDown(thumb, {key: 'PageDown', code: 34});

    // Should render the slider with updated value
    expect(slider.asFragment()).toMatchSnapshot();
  });

  test('calls onChange with value updates', () => {
    const onChange = jest.fn();

    const slider = renderWithTheme(StatefulSlider, {
      onChange,
      values: [50],
      min: 25,
      max: 75,
      step: 1,
    });

    // Simulate pressing page up to raise value by 10
    const thumb = slider.getByRole('slider');
    fireEvent.keyDown(thumb, {key: 'PageUp', code: 33});

    // Should call onChange with new value
    expect(onChange).toHaveBeenCalledWith([60]);
  });
});
