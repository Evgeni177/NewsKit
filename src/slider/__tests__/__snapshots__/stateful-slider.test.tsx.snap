// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatefulSlider holds the slider values in state 1`] = `
<DocumentFragment>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  position: relative;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  box-sizing: border-box;
  background-color: #f0f0f0;
  border-radius: 20rem;
  margin: calc(32px/2 - 12px/2) calc(32px/2);
  height: 12px;
  width: 100%;
}

.emotion-1:disabled {
  background-color: #f0f0f0;
}

.emotion-0 {
  background-color: #0A68C1;
  border-color: #0A68C1;
  border-width: 1px;
  border-radius: 50%;
  height: 32px;
  width: 32px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
}

.emotion-0 svg {
  fill: #9c9c9c;
  color: #9c9c9c;
}

.emotion-0:hover:not(:disabled) {
  background-color: #02509B;
}

.emotion-0:active:not(:disabled) {
  background-color: #01396E;
}

.emotion-0:active:not(:disabled) svg {
  fill: #535353;
  color: #535353;
}

.emotion-0:disabled {
  border-style: none;
}

.emotion-0:disabled svg {
  fill: #9c9c9c;
  color: #9c9c9c;
}

<div
    class="emotion-2"
    data-testid="slider"
  >
    <div
      aria-orientation="horizontal"
      aria-valuemax="75"
      aria-valuemin="25"
      class="emotion-1"
      data-testid="slider-track"
      values="40"
    >
      <div
        aria-label="slider"
        aria-valuemax="75"
        aria-valuemin="25"
        aria-valuenow="40"
        class="emotion-0"
        data-testid="slider-thumb"
        draggable="false"
        role="slider"
        style="position: absolute; z-index: 0; cursor: grabbing; user-select: none; transform: translate(NaNpx, NaNpx);"
        tabindex="0"
        values="40"
      />
    </div>
  </div>
</DocumentFragment>
`;
